doc/source/user_guide/ug_implementation.rst

```diff
@@ -367,6 +420,122 @@ or southern hemispheres, respectively. Special constants (``spval`` and
+
+.. _tab-gridsys:
+
+.. table:: Grid System and Type Definitions
+   :align: center
+
+   +--------------+----------------+----------------+----------------+
+   | grid system  |   thermo grid  | u dynamic grid | v dynamic grid |
+   +==============+================+================+================+
+   |     A        |       T        |       T        |       T        |
+   +--------------+----------------+----------------+----------------+
+   |     B        |       T        |       U        |       U        |
+   +--------------+----------------+----------------+----------------+
+   |     C        |       T        |       E        |       N        |
+   +--------------+----------------+----------------+----------------+
+   |     CD       |       T        |       N+E      |       N+E      |
+   +--------------+----------------+----------------+----------------+
+
+For all grid systems, thermodynamic variables are always defined on the ``T`` grid for the model and 
+model forcing/coupling fields.  However, the dynamics u and v fields vary.
+In the ``CD`` grid, there are twice as many u and v fields as on the other grids.  Within the CICE model,
+the variables ``grid_ice_thrm``, ``grid_ice_dynu``, ``grid_ice_dynv``, ``grid_atm_thrm``, 
+``grid_atm_dynu``, ``grid_atm_dynv``, ``grid_ocn_thrm``, ``grid_ocn_dynu``,  and ``grid_ocn_dynv`` are
+character strings (``T``, ``U``, ``N``, ``E`` , ``NE``) derived from the ``grid_ice``, ``grid_atm``, 
+and ``grid_ocn`` namelist values.
+
+The CICE model has several internal methods that will interpolate (a.k.a. map or average) fields on 
+(``T``, ``U``, ``N``, ``E``, ``NE``) grids to (``T``, ``U``, ``N``, ``E``).  An interpolation
+to an identical grid results in a field copy.  The generic interface to this method is ``grid_average_X2Y``,
+and there are several forms.
+
+.. code-block:: fortran
+
+      subroutine grid_average_X2Y(type,work1,grid1,work2,grid2)
+        character(len=*)    , intent(in)  :: type           ! mapping type (S, A, F)
+        real (kind=dbl_kind), intent(in)  :: work1(:,:,:)   ! input field(nx_block, ny_block, max_blocks)
+        character(len=*)    , intent(in)  :: grid1          ! work1 grid (T, U, N, E)
+        real (kind=dbl_kind), intent(out) :: work2(:,:,:)   ! output field(nx_block, ny_block, max_blocks)
+        character(len=*)    , intent(in)  :: grid2          ! work2 grid (T, U, N, E)
+
+where type is an interpolation type with the following valid values,
+
+type = ``S`` is a normalized, masked, area-weighted interpolation
+
+.. math:: 
+   work2 = \frac{\sum_{i=1}^{n} (M_{1i}A_{1i}work1_{i})} {\sum_{i=1}^{n} (M_{1i}A_{1i})}
+
+type = ``A`` is a normalized, unmasked, area-weighted interpolation
+
+.. math:: 
+   work2 = \frac{\sum_{i=1}^{n} (A_{1i}work1_{i})} {\sum_{i=1}^{n} (A_{1i})}
+
+type = ``F`` is a normalized, unmasked, conservative flux interpolation
+
+.. math:: 
+   work2 = \frac{\sum_{i=1}^{n} (A_{1i}work1_{i})} {n*A_{2}}
```

Comment:
Is the documentation available in a readable form somewhere?  I'm having trouble understanding {1i} and {2}...